hw_7/DOM.js

```diff
@@ -0,0 +1,301 @@
+*/
+
+
+// Вариант № 2
+//ul_td.innerHTML = getItem('data_myUl');
+
+let myLi = [];
+let htmlStr = '';
+let maxId = getMax(myLi);
+
+
+if (getItem('data_myUl')){
+   let myLi = getItem('data_myUl');
+   render(myLi);
+}
+
+function getMax(myObj) {
+    let tempPer = 0;
+    for (let i = 0; i < myObj.length; i++) {
+        tempPer = Math.max(tempPer, myObj[i].id);
+    }
+    return tempPer;
+}
+
+
+function addLi(text) {
+    let myTemp = {
+        id: ++maxId,
+        text: text,
+        data: new Date().toLocaleString(),
+        isCheck: false,
+    }
+
+    myLi.push(myTemp);
+
+    htmlStr += addLihtml(maxId, text, myTemp.data, myTemp.isCheck);
+}
+function addLihtml(id, text, data, isCheck) {
+    return `<li id="${id}">
+                        <div class="todo-li">
+                            <div>
+                                <input type="checkbox" class="check" ${isCheck ? 'checked' : ''}>
+                                <div><span class="doToText">${text}</span></div>
+                                <button class="close">x</button>
+                            </div>
+                            <div>
+                                <div><span class="date">${data}</span></div>
+                            </div>
+                        </div>
+                    </li>`
+}
+
+function delLi(id) {
+    myLi = myLi.filter(item => item.id != id);
+    render(myLi);
+}
+
+function render(myLi) {
+    htmlStr = '';
+    if (myLi.length) {
+        //return ul_td.innerHTML = '';
+        for (let i = 0; i < myLi.length; i++) {
+        htmlStr += addLihtml(myLi[i].id, myLi[i].text, myLi[i].data, myLi[i].isCheck);
+         }
+    }   
+    ul_td.innerHTML = htmlStr;
+}
+function isCheck(elem) {
+    let tempPer = elem.closest('li').id;
+    for (let i = 0; i < myLi.length; i++) {
+        if (myLi[i].id == tempPer) {
+            myLi[i].isCheck = elem.checked;
+            break;
+        }
+    }
+    render(myLi);
+}
+
+function onAddLi(event) {
+    if (event.target.className === 'add') {
+        let val = document.querySelector('.inputText').value;
+        addLi(val);
+    }
+    document.querySelector('.inputText').value = '';
+    render(myLi);
+    setItem('data_myUl', JSON.stringify(myLi));
+}
+
+function onDelAllLi(event) {
+    if (event.target.className === 'deleteAll') {
+        ul_td.innerHTML = '';
+    }
+    htmlStr = '';
+    myLi.length = 0;
+    maxId = 0;
+    setItem('data_myUl', JSON.stringify(myLi));
+}
+function onDelLi(event) {
+    if (event.target.className === 'close') {
+        delLi(event.target.closest('li').id);
+    }
+    render(myLi);
+    setItem('data_myUl', JSON.stringify(myLi));
+}
+
+function onChecked(event) {
+    if (event.target.className === 'check') {
+        isCheck(event.target);
+    }
+    setItem('data_myUl', JSON.stringify(myLi));
+}
+
+function onStor(event) {
+    if (event.key === 'data_myUl') {
+        ul_td.innerHTML = htmlStr;
```

Comment:
не понял что тут происходит)
но думаю
что тут надо взять значение и поместить его в массив и перерисовать список
```suggestion
        myLi  = JSON.parse(event.newValue);
        render(myLi);
```